### PR TITLE
Remove sphinx dependency and integrate with skygear-doc-tools

### PR DESCRIPTION
**Pull request by howawong**
Connects #169

### CODE REVIEW

**Comment by rickmak** [comment]
Remove sphinx dependency is good things. However, move all the doc logic into other repo is seem a bad organization to me.

I think the grouping:

1. in py-skygear, where the inline doc is written, can be generate the doc using standard tools. And since the tooling and writing format are couple, putting them into two repos does not see right to me.
1. I think skygear-doc-tools is cool to automate the doc building process. But I think its responsibility should only restricted to composite of command to build, upload, cache invalidation, release management.

I propose to keep the `python setup.py doc` command working with inline import of sphinx. And make a requirement-dev.txt for sphinx dependency.

If you find the above grouping is not aligned in iOS/Android., I think we should try to discuss it and align it.

**Comment by howawong** [comment]
FYI

1. Tooling and Writing Format are not coupled. The formatting style is Google Ptython Style Guide, https://google.github.io/styleguide/pyguide.html. Just sphinx is currently the most used documentation tool, does not mean there is a coupling. If you really bother, you can just check Doxygen and https://pypi.python.org/pypi/doxypypy/0.8.8.6  (Though it is not really maintained.) And rST is a well known format.

2. inline import is not preferred as it just makes dependency messy. And then if there is another project requiring document generation, developer copies and pastes inline import to another project? I don't think that's a right approach.

3. If `requirement-dev.txt` is valid, why copying few lines to every project, but not just one line of git+https in pip? And why a developer needs to touch another requirment-dev.txt if he just need a doc ready in travis.

4. Copying and Pasting already exists in core SDK and chat SDK, such thing is against DRY and a disaster to me.

**Comment by rickmak** [comment]
I think the solution can be you add the doc generation packages to `extras_require`. In `.travis.yml`, run `pip install .[doc]`. You can still have your repeating code in doc-tools while the docs generation is still idiomatic to most of the python developer.

TLDR;

<hr>

1. Yes, I am not referring to the writing format. But generating the doc is more about just writing format. For instance, how the table of content is structure. I think there is no common format? (correct me if I am wrong.)

I think if we only intended to support Sphinx as the target generating doc tool while writing the doc, making it explicit helps.

Also, you checked in a `doc/conf.py` which is implying a sphinx as the target.

2. inline import is not preferred, but it is acceptable in some case. Take a look at the existing `PyTest` class at `setup.py` and none complaint. And it comes with setuptools default.

I think before we arguing and invent our own approach, we should take a reference to how is the popular project setup. For example, https://github.com/zzzeek/sqlalchemy/ and https://github.com/Pylons/pyramid are putting doc generating command into Makefile. We might consider this approach.

For the copying and pasting, I think you are referring to we need doc generation tools for all our project. So it will be making the doc generation code appear everywhere.

### Summary

The original issue
- `python setup.py install` should not install Sphinx

What howa want
- Less copy & paste on the source about doc generation

What rickmak want
- keep the doc generation python user discoverable within the directory and idiomatic
  - best if `python setup.py doc` works(optional)
  - `sphinx-build` will works in doc folder
  - installing `skygear doc tools` is not required if I only generating the doc by sphinx-build
  - make the dependency specification appear in either `requirement.txt` or as `extras_require` in `setup.py`.

File: setup.py [hunk 1]
```diff
@@ -51,23 +48,28 @@ def run(self):
...
-                            '-R', metadata.version,
-                            '-o', temp_dir,
-                            path.join('..', 'skygear')])
-        for rst in glob.glob(r'*.rst') + ['conf.py']:
-            print(rst)
-            copyfile(rst, path.join(temp_dir, rst))
-        sphinx.main(['', temp_dir, '_build'])
+        mkdir(temp_dir)
+        environ['SPHINX_APIDOC_OPTIONS'] = 'members'
+        system('sphinx-apidoc %s -H %s -A %s -V %s -R %s -o %s %s' %
```
Comment (cheungpat) [inline setup.py]: Is there any material difference between calling sphinx binaries via `system()` and running the binaries using `Makefile`?

by the way, have you considered using `setup_requires`? https://stackoverflow.com/questions/21149035/autogenerate-documentation-for-python-project-using-setuptools

**Comment by rickmak** [comment]
Please try to squash out the testing commit. https://m.oursky.com/i-care-about-commit-quality-so-should-you-19b7fac8a408

**Comment by howawong** [comment]
@rickmak  will squash if no other comment.

File: setup.py (hunk 1 above)
Comment (rickmak) [reply setup.py]: No, No.

Both `Makefile` and py system is fine for me.
But here I wonder why don't use `sphinx` module here but system. Avoid inline import?

No, my intuition makes me think `doc` is extra, not required for `setup`. i.e. I can setup the project and fork it without modifying the doc.

Comment (howawong) [reply setup.py]: Why using Makefile if you are not making use of dependency ?

Please check
https://pip.pypa.io/en/stable/reference/pip_install/

Beware of the setup_requires keyword arg in setup.py. The (rare) packages that use it will cause those dependencies to be downloaded by setuptools directly, skipping pip's hash-checking. If you need to use such a package, see Controlling setup_requires.

Comment (howawong) [reply setup.py]: FYI, after talking to @rickmak, keeping Makefile would be better as it is used across majority of the projects. I will improve the doc-tools and update very soon.

**Comment by rickmak** [comment]
Merge via 2a90e24
